lib/puppet/defaults.rb

```diff
@@ -32,6 +32,41 @@ def self.valid_file_checksum_types
       %w[sha256 sha256lite sha384 sha512 sha224 sha1 sha1lite md5 md5lite mtime ctime]
   end
 
+  def self.log_ca_migration_warning
+    urge_to_migrate = <<-UTM
+The cadir is currently in your puppet/ssl directory. This location will be
+deprecated and removed in a future version of puppet. Please run the the
+puppetserver ca tool to migrate it out to the puppetserver directory. Use
+`puppetserver ca migrate --help` for more info.
+UTM
+    Puppet.warn_once('deprecations',
+                     'CA migration message',
+                     urge_to_migrate,
+                     :default,
+                     :default)
+  end
+
+  def self.default_cadir
+    old_ca_dir = "#{Puppet[:ssldir]}/ca"
+    new_ca_dir = "/etc/puppetlabs/puppetserver/ca"
+
```

Comment:
Assuming the CA can never be on Windows, then maybe `return old_ca_dir if Puppet::Util::Platform.windows?` so that we skip the symlink detection and don't return a posix path `/etc/puppetlabs/puppetserver/ca`? Although the agent never calls `Puppet[:cadir]`, it would end up in the output of `puppet config print all`.

Reply:
hmmm, I'm reticent to return the old value; is it ok if we return an empty string or nil? I do agree that returning before symlink detection is good on Windows though.

Reply:
I think nil is fine, we do that for some others like `priority`

Reply:
I did end up going with empty string here; I think because of the way the hooks work, using `nil` would end up with a method not found error when the hook ran, because it calls `start_with?`.